### PR TITLE
(PA-538) Add committers and contributing guidelines

### PR DESCRIPTION
**Pull request by MosesMendoza**
This PR initially adds the COMMITTERS.md and CONTRIBUTING.md files from the github.com/puppetlabs/puppet project as a baseline for ones to use for Puppet Agent, and then modifies them for use in Puppet Agent.

In COMMITTERS.md file copied from github.com/puppetlabs/puppet for Puppet Agent, references to Puppet are replaced with Puppet Agent. Additionally, notice of a 'security' branch is removed, because Puppet  Agent does not have a security branch nor is this part of the current workflow. We also remove reference to YARD docs as the Puppet Agent project does not have these.

In the CONTRIBUTING.md file, references to Puppet core vs modules are removed, as they are inapplicable. Triage meeting/notes info is removed as we don't have a meeting like that scheduled at the time of this PR. In general though, everything else applies.

### CODE REVIEW

**Comment by MosesMendoza** [comment]
cc @puppetlabs/agent-team for review

File: COMMITTERS.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+---
+
+This section aims to provide guidelines for being a good commit citizen by
+paying attention to our automated build tools.
+
+ * Don’t push on a broken build.  (A broken build is defined as a failing job
+   in the [Puppet Agent](https://jenkins.puppetlabs.com/view/puppet-agent/)
+   page.)
+ * Watch the build until your changes have gone through green Update the ticket
+ * status and target version.  The target version field in
```
Comment (Magisus) [inline COMMITTERS.md]: This looks like a weird newline in the wrong spot, but maybe it'll be fine after markdown parsing?

Comment (MosesMendoza) [inline COMMITTERS.md]: I think you're right - I think that's a mistake from the word-wrap. Will update

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    based on a description rather than an example.  This patch fixes the
+    problem by making the example concrete and imperative.
+
+    The first line is a real life imperative statement with a ticket number
+    from our issue tracker.  The body describes the behavior without the patch,
+    why this is a problem, and how the patch fixes the problem when applied.
+````
+
+* Make sure you have added the necessary tests for your changes.
+* Run _all_ the tests to assure nothing else was accidentally broken.
```
Comment (stahnma) [inline CONTRIBUTING.md]: How does one run tests for puppet-agent?

File: COMMITTERS.md (hunk 1 above)
Comment (MosesMendoza) [inline COMMITTERS.md]: Updated

File: CONTRIBUTING.md (hunk 2 above)
Comment (MosesMendoza) [inline CONTRIBUTING.md]: Oh that's interesting. Do we really not even have specs for puppet-agent? I assumed there was some part of https://jenkins.puppetlabs.com/view/puppet-agent/view/manual/view/stable/ that actually did some testing, but it appears not..

Comment (MosesMendoza) [inline CONTRIBUTING.md]: updated to remove that line

File: COMMITTERS.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+We would like to make it easier for community members to contribute to Puppet
+Agent using pull requests, even if it makes the task of reviewing and
+committing these changes a little harder.  Pull requests are only ever based on
+a single branch, however, we maintain more than one active branch.  As a result
+contributors should target their changes at the master branch. This makes the
+process of contributing a little easier for the contributor since they don't
+need to concern themselves with the question, "What branch do I base my changes
+on?"  This is already called out in the
+[CONTRIBUTING.md](https://goo.gl/XRH2J).
```
Comment (geoffnichols) [inline COMMITTERS.md]: It looks like this points back to the puppet repo. Maybe we should update or remove this?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    The first line is a real life imperative statement with '(doc)' in
+    place of what would have been the ticket number in a
+    non-documentation related commit. The body describes the nature of
+    the new documentation or comments added.
+````
+
+## Submitting Changes
+
+* Sign the [Contributor License Agreement](http://links.puppet.com/cla).
```
Comment (geoffnichols) [inline CONTRIBUTING.md]: We should probably confirm the CLA bot is watching this repo.

Comment (MosesMendoza) [inline CONTRIBUTING.md]: +1. I emailed @jeffweiss as I think? he adds them? At least according to confluence docs..

File: COMMITTERS.md (hunk 3 above)
Comment (MosesMendoza) [inline COMMITTERS.md]: thank you! Updated

File: CONTRIBUTING.md (hunk 4 above)
Comment (MosesMendoza) [inline CONTRIBUTING.md]: ok, according to @jeffweiss this repo is already monitored by cla bot

Comment (MosesMendoza) [inline CONTRIBUTING.md]: CLA bot has commented to confirm

**Comment by MosesMendoza** [comment]
ping @puppetlabs/agent-team for merge

**Comment by jtappa** [comment]
In docs, we always refer to Puppet agent with a lowercase 'a'.

**Comment by MosesMendoza** [comment]
Hey @jtappa - just to make sure I understand your feedback - you're requesting that anywhere "Puppet Agent" occurs in these two files, it's replaced with "Puppet agent" ?

**Comment by jtappa** [comment]
@MosesMendoza Yup, that's right. I just didn't want to barrage you with a bunch of comments on each one.

**Comment by MosesMendoza** [comment]
Thanks @jtappa. Updated

**Comment by Magisus** [comment]
👍

**Pull request merged**